clients/src/main/java/org/apache/kafka/clients/consumer/RetriableCommitFailedException.java

```diff
@@ -16,21 +16,16 @@
  */
 package org.apache.kafka.clients.consumer;
 
-import org.apache.kafka.common.errors.RetriableException;
+import org.apache.kafka.common.errors.ApiException;
 
-public class RetriableCommitFailedException extends RetriableException {
+public class RetriableCommitFailedException extends ApiException {
 
     private static final long serialVersionUID = 1L;
 
-    public static RetriableCommitFailedException withUnderlyingMessage(String additionalMessage) {
-        return new RetriableCommitFailedException("Offset commit failed with a retriable exception. " +
-                "You should retry committing the latest consumed offsets. " +
-                "The underlying error was: " + additionalMessage);
-    }
-
     public RetriableCommitFailedException(Throwable t) {
-        super("Offset commit failed with a retriable exception. You should retry committing " +
-                "the latest consumed offsets.", t);
+        super("Offset commit failed with a retriable exception. This is usually because the consumer is no longer " +
+            "part of a group or is participating a rebalance right now. You should first call poll to complete " +
```

Comment:
`the consumer is no longer part of a group` is a bit ambiguous, it sounds like this will be thrown if a consumer falls out of the group or hasn't yet joined, but in those cases we just throw `CommitFailedException`. Can we rephrase this somehow? When might this be thrown for reasons besides a rebalance in progress?